nipype/pipeline/engine/nodes.py

```diff
@@ -441,6 +443,7 @@ def run(self, updatehash=False):
             for outdatedhash in glob(op.join(self.output_dir(), '_0x*.json')):
                 os.remove(outdatedhash)
 
+        self._get_hashval()
```

Comment:
@oesteban - is there a specific reason for calling this here?

Reply:
Yes, it was being called within `is_cached`. But now, `is_cached` exists fast if the result file is not there, etc. So, if the node's folder or the result file did not exist, flow reached here without generating the hashval.

Reply:
thank you.

Reply:
I think it'd be good to add a comment to this effect.

Reply:
Added